### PR TITLE
added password reset form

### PR DESCRIPTION
**Pull request by liewshanhong**
implemented password reset feature

### CODE REVIEW

**Comment by SanaBob** [comment]
Implemented email verification: register -> still cannot login/upload resume -> verify email will be sent to the registered email with a verification link -> after verifying, have access to all features

File: skills_DS/base/views.py [hunk 1]
```diff
@@ -2,17 +2,68 @@
...
+            mail_subject = 'Activate your account.'
+
+            message = render_to_string('base/email_template.html', {
+                'user': user,
+                'domain': current_site.domain,
+                'uid': urlsafe_base64_encode(force_bytes(user.pk)),
+                'token': token_generator.make_token(user),
+            })
+            to_email = form.cleaned_data.get('email')
+            send_mail(mail_subject, message, 'cosc448atesting@gmail.com', [to_email])
```
Comment (brianzhouzc) [inline skills_DS/base/views.py]: We should probably store the email address somewhere else other than hard coding it.

**Review by brianzhouzc — APPROVED** [review]
I don't see any problem with the code other than the credentials I mentioned in another pull request. Good to merge for now.

File: skills_DS/base/views.py (hunk 1 above)
Comment (brianzhouzc) [reply skills_DS/base/views.py]: There's a setting option we are already using, so we should use that instead.

**Comment by SanaBob** [comment]
+ resending verification email on login if user has not verificated. (authenticate method was updated by django, so I had to implement a lot of ifs)

**Comment by OKThomas1** [comment]
Credentials were invalid, but I tested it with some other credentials and everything worked as expected.

**Review by OKThomas1 — APPROVED** [review]

**Pull request merged**